watcher/lib/influxdb/influxdb.go

```diff
@@ -155,23 +233,92 @@ func (c *Client) execQuery(query string) error {
 	return nil
 }
 
+func (c *Client) setupRetentionPolicies() error {
+	queries := []string{
+		fmt.Sprintf(createRetentionPolicyQuery, constants.InfluxDBRetentionPolicy,
+			constants.InfluxDBDatabase, constants.DurationDefault) + " default",
+		fmt.Sprintf(createRetentionPolicyQuery, constants.RetentionMedium, constants.InfluxDBDatabase,
+			constants.DurationMedium),
+		fmt.Sprintf(createRetentionPolicyQuery, constants.RetentionLong, constants.InfluxDBDatabase,
+			constants.DurationLong),
+	}
+	for _, query := range queries {
+		log.WithField("query", query).Info("Setup retention policies query.")
+
+		err := c.execQuery(query)
+		if err != nil {
+			if trace.IsAlreadyExists(ConvertInfluxDBError(err)) {
+				log.Info("Retention policy already exists with different attributes. Skipping it.")
+				continue
+			}
+			return trace.Wrap(err)
+		}
+	}
+	return nil
+}
+
 // ConvertInfluxDBError converts error from InfluxDB query results
 func ConvertInfluxDBError(err error) error {
-	if strings.Contains(err.Error(), "retention policy already exists") {
+	if strings.Contains(err.Error(), "already exists") {
```

Comment:
Is there InfluxDB-specific error types or text-based tests are the only way to differentiate errors?
Also would be helpful if methods dealing with InfluxDB APIs directly used `ConvertError` so callers would not need to.

Reply:
Good point. I've found influxdb errors in server code. Can reuse them here.